packages/rspack/src/util/bindingVersionCheck.ts

```diff
@@ -0,0 +1,18 @@
+import { execSync } from "child_process";
+
+function getVersion(packageName: string): string {
+	const command = `npm show ${packageName} version`;
+	return execSync(command).toString().trim();
+}
+
+export const bindingVersionCheck = () => {
+	const coreVersion = getVersion("@rspack/core");
+	const bindingVersion = getVersion("@rspack/binding");
+	if (coreVersion !== bindingVersion) {
+		throw new Error(
```

Comment:
This error should be propagated property instead of throwing here directly.

Reply:
> This error should be propagated property instead of throwing here directly.

I don't really understand why， can you explain？

Reply:
Errors thrown this way is what users cannot expect and it would be better to use the node callback instead of current way.